astria api

burgalon opened the pull request.

Tristoun reviewed (changes requested):
- Catch errors by raising a ProviderException if the response status is not good, instead of using raise for status.
- Utilize strings for the provider_job_id.
- Change the url when create a job with image__generate.
- Resolve the infinite loop in tune creation caused by multiple "files."
- Allow users to use a base_tune_id other than 690204.
- Enable users to create entities other than just "man" or "woman."
- Resolve issue with **data, doens't work because of differences between names

burgalon commented:
1) Pending 
2) Provider_job_id is used indeed as much as I can see
3) Not sure what you mean by that
4) Can you be more specific to the line of where you're seeing an infinite loop posibility?
5) added base_tune_id
6) class_name is indeed an argument and allows any class, and is not limited to man/woman
7) Fixed

Tristoun commented on edenai_apis/apis/astria/astria_api.py:
```diff
@@ -26,30 +31,81 @@ def __init__(self, api_keys: Dict = {}) -> None:
+
+        response = requests.post(f"{self.url}tunes", data=data, files=files, headers=self.headers)
+        response.raise_for_status()
+        return AsyncLaunchJobResponseType(provider_job_id=response.json()["id"])
```
provider_job_id is a string not an integer

burgalon replied:
fixed

Tristoun commented on edenai_apis/apis/astria/astria_api.py:
```diff
@@ -26,30 +31,81 @@ def __init__(self, api_keys: Dict = {}) -> None:
+        }
+        files = []
+        if input_image:
+            files.append((f"tune[prompts_attributes][{i}][input_image]", load_image(input_image)))
```
i is not defined

burgalon replied:
fixed

Tristoun commented on edenai_apis/apis/astria/astria_api.py:
```diff
@@ -26,30 +31,81 @@ def __init__(self, api_keys: Dict = {}) -> None:
+        if input_image:
+            files.append((f"tune[prompts_attributes][{i}][input_image]", load_image(input_image)))
+
+        response = requests.post(f"{self.url}/tunes/{project_id}", headers=self.headers, data=data)
```
this link doesn't work use this link instead
https://api.astria.ai/tunes/project_id/prompts

burgalon replied:
fixed

Tristoun commented on edenai_apis/apis/astria/astria_api.py:
```diff
@@ -26,30 +31,81 @@ def __init__(self, api_keys: Dict = {}) -> None:
+        }
+        for image in files:
+            image_data = load_image(image)  # Assuming image is a file path
+            files.append(("tune[images][]", image_data))
```
Append to files which create a problem because files list already exist

burgalon replied:
indeed, and that's a good thing :)

burgalon replied:
maybe i'm missing something... but that should be good

Tristoun commented on edenai_apis/apis/astria/astria_api.py:
```diff
@@ -26,30 +31,81 @@ def __init__(self, api_keys: Dict = {}) -> None:
+            status="succeeded" if data['trained_at'] else "pending",
+            provider_job_id=provider_job_id,
+            original_response=data,
+            standardized_response=GenerationFineTuningGenerateImageAsyncDataClass(**data),
```
**data can't work because difference between names

burgalon replied:
fixed

Tristoun commented on edenai_apis/apis/astria/astria_api.py:
```diff
@@ -26,30 +31,81 @@ def __init__(self, api_keys: Dict = {}) -> None:
+            'prompt[text]': prompt,
+            'prompt[negative_prompt]': negative_prompt,
+            'prompt[num_images]': num_images,
+            'prompt[face_swap]': True,
```
If I want to create something else other than human return an error (try to create a dog)

burgalon replied:
added args

burgalon commented:
added some fixes
